src/compute/src/render/context.rs

```diff
@@ -1064,44 +1069,48 @@ where
     fn specialized_arrange(
         name: &String,
         oks: Collection<S, Row, i64>,
-        key: &Vec<MirScalarExpr>,
-        thinning: &Vec<usize>,
+        key: Vec<MirScalarExpr>,
+        thinning: Vec<usize>,
     ) -> (MzArrangement<S>, Collection<S, DataflowError, i64>) {
-        // Catch-all: Just use RowRow.
-        let (oks, errs) = oks
-            .map_fallible::<CapacityContainerBuilder<_>, CapacityContainerBuilder<_>, _, _, _>(
-                "FormArrangementKey",
-                specialized_arrangement_key(key.clone(), thinning.clone()),
+        let (oks, errs) =
+            oks.inner
+                .unary_fallible::<ColumnBuilder<((Row, Row), S::Timestamp, i64)>, _, _, _>(
+                    Pipeline,
+                    "FormArrangementKey",
+                    move |_, _| {
+                        Box::new(move |input, ok, err| {
```

Comment:
This went from being a `map_fallible` to something more complicated. I think it would benefit from a comment explaining that (if correct) this is implementing a `map_fallible`, but it appears complicated in order to accommodate containers of rows.

Reply:
The comment at the top of the method says weird stuff about specialization and bit lengths. This no longer seems to be accurate, and perhaps that is a fine place to leave the explanation about what the method does. Ideally it would describe the arguments `key` and `thinning` specifically, as I believe these are generally complicated (thinning, for sure). Not necessarily how they work, but if it is exactly the thing that has a clear meaning where created (the thinning and permutation stuff) perhaps a reference to it.

Reply:
Updated the doc comment and added an inline comment.